File: Easy/twoSum.js
//two sum
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const number_map = new Map();
    for(let i = 0; i < nums.length; i++){
        diff = target - nums[i];
        if(number_map.has(diff)){
            return [i,number_map.get(diff)];
        }
    
        number_map.set(nums[i],i);
    }
};

File: Easy/twoSum.py
#two sum
class Solution(object):
    def twoSum(self, nums, target):
        number_map = {}
        
        for x,num in enumerate(nums):
            diff = target - num
            if(diff in number_map):
                return [x,number_map[diff]]
            number_map[num] = x

        return None